### PR TITLE
Add clearing of obs location

### PR DESCRIPTION
**Pull request by yuliadub**
attempt to solve https://github.com/NWACus/avy/issues/656. This changes the behavior a bit, where to select a location you now push a "check" location and "x" to clear it. The part that is not perfect is clearing the coordinates fully, you can spot that if you clear and then come back to set it again - the point will briefly be visible at the previous location if there was one. 

https://nwactechcommittee.slack.com/archives/C051KR96B2L/p1711086267002079?thread_ts=1711086244.167219&cid=C051KR96B2L

### CODE REVIEW

File: components/form/LocationField.tsx [hunk 1]
```diff
@@ -72,6 +83,7 @@ export const LocationField = React.forwardRef<RNView, LocationFieldProps>(({name
   const onPress = useCallback(
```
Comment (stevekuznetsov) [inline components/form/LocationField.tsx:83]: I think with the current implementation if you were to clear your location, the `field` would still have the data in there and you could still submit the observation with the previously chosen point. Do you think we could get clearing behavior by updating the form field to be empty?

Comment (yuliadub) [reply components/form/LocationField.tsx]: ha, not sure why overcomplicated it so much originally. let me know if this looks more like it to you!

**Comment by stevekuznetsov** [comment]
Looks good!

**Pull request merged**